updater/lib/dependabot/update_files_command.rb

```diff
@@ -129,5 +134,20 @@ def handle_parser_error(error)
       )
     end
     # rubocop:enable Metrics/AbcSize, Layout/LineLength, Metrics/MethodLength
+
+    def handle_dependency_file_not_parseable_error(error)
+      error_details = Dependabot.updater_error_details(error)
+
+      service.record_update_job_error(
+        error_type: T.must(error_details).fetch(:"error-type"),
+        error_details: T.must(error_details)[:"error-detail"]
+      )
+      return unless Experiments.enabled?(:record_update_job_unknown_error)
+
+      service.record_update_job_unknown_error(
+        error_type: T.must(error_details).fetch(:"error-type"),
```

Comment:
any reason to record this as an unknown error again?

Reply:
Mostly because that's what's being done with the same error in this same file a few lines earlier. I'm not completely sure what the rules are for this categorisation so I just went with the precedent.